### DISCUSSION TITLE
Insert different variables depending on the condition

### DISCUSSION THREAD

**Discussion by IvanIsak2000** [Textualize/textual#2855] [General]
For example, I have 5 different lines of text and I want one line to be output through Label from some condition.

Of course, I could write 5 different classes, each class would produce its own string, but I want to make a general class, and so that the `compose` function of this class receives a variable and prints it, how can I do this?

**Comment by davep** [comment]
It's not really clear to be what aspect of Textual you're asking about here, I could read this a few ways. Do you think you could try and reword the question, perhaps with some sort of illustrative example?

**Reply by IvanIsak2000** [reply]
Okay, I want to create my own tui game, in this game you need to select a character, let's say there is a Warrior character and the player selects it, and I want the logo of this character to be drawn.

To do this, I want to make a general class for displaying the character's logo and, depending on the passed argument, display the character's logo. How can I do it ?

For example:
How I want:
```bash
class RenderCharacterLogo() : 
def compose(self, ????):
    yield Label(character_logo)


get_user_character = get_character()
character = get_user_character.run()

if character == 'Warrior':
?????

```

Worse option:

```bash 
class RenderWarriorLogo():
def compose(self):
    yield Label('Warrior')

class RenderPaladinLogo():
def compose(self):
    yield Label('Paladin')

get_user_character = get_character()
character = get_user_character.run()

if character == 'Warrior':
    warrior = RenderWarriorLogo()
    warrior.run()

elif character == 'Paladin':
    paladin = RenderPaladinLogo()
    paladin.run()

```

**Reply by davep** [reply]
I'm still finding it difficult to really grasp what aspect of Textual it is you're asking about, perhaps down to me not quite appreciating what sort of data you're working with and how you'd want the user to interact with your application. But also, from what you're showing here, and other questions you've asked, I sense that you might be early in your journey with learning Textual, and perhaps have misunderstood how applications and screens and widgets etc all work together (for example, it would be *very* unusual to be creating lots of different applications and running them individually, as you seem to be indicating above).

Have you [worked through the tutorial step-by-step](https://textual.textualize.io/tutorial/) and had a read through the [guide](https://textual.textualize.io/guide/). If not, I'd very much encourage it. I suspect it will help with understanding how you can use Textual to build your game's interface.

**Reply by IvanIsak2000** [reply]
Thanks for your reply! Indeed, I will study the documentation again

**Reply by davep** [reply]
Feel free to follow up with any questions you have if anything isn't clear.
